### ISSUE TITLE
python bindings seem confused for 4.0.0 build

### ISSUE THREAD

**Issue by amckinstry** [openpmix/openpmix#2003]
This is on Debian Unstable Linux, with pmix 4.0.0

For reference, we do two builds - a static build to get .a files, and then a shared build - shared is the default used by applications, but the .a libs are there for users to build optimised.

This means we do builds in subdirectories, `debian/shared-build `and   `debian/static-build`.
This works but when the `--enable-python-bindings` flag is used, it fails, as it presumes that 
`$TOPDIR/include` == `$BUILDDIR/include".

This is fixed in current Debian builds with a hack that copies all the contents of the include files to each other, between the config and build stages.

It would be good to get a fix for this included in any 4.0.1 release.

**Comment by jjhursey** [comment]
I have a fix in PR #2004 if you want to give it a try.

I tested with:
```
mkdir -p debian/shared-build
mkdir -p debian/static-build

cd debian/shared-build/
../../configure --prefix=/tmp/pmix-shared --enable-shared --disable-static --with-libevent=$HOME/local/libevent --with-hwloc=$HOME/local/hwloc-2.1 --enable-python-bindings
make -j 20
make -j 20 install

cd ../static-build
../../configure --prefix=/tmp/pmix-static --disable-shared --enable-static --with-libevent=$HOME/local/libevent --with-hwloc=$HOME/local/hwloc-2.1 --enable-python-bindings
make -j 20
make -j 20 install

cd ../../
../../configure --prefix=/tmp/pmix-default --with-libevent=$HOME/local/libevent --with-hwloc=$HOME/local/hwloc-2.1 --enable-python-bindings
make -j 20
make -j 20 install
```

**Comment by jjhursey** [comment]
Cherry Pick for `v4.0.1` in PR #2012
